Kudu.Core/Deployment/Generator/CustomBuilder.cs

```diff
@@ -16,11 +18,27 @@ public CustomBuilder(IEnvironment environment, IDeploymentSettingsManager settin
 
         public override Task Build(DeploymentContext context)
         {
+            string commandFullPath = _command;
             var tcs = new TaskCompletionSource<object>();
             context.Logger.Log("Running custom deployment command...");
 
             try
             {
+                if (!OSDetector.IsOnWindows())
+                {
+                    if (commandFullPath.StartsWith("."))
+                    {
+                        string finalCommandPath = Path.GetFullPath(Path.Combine(RepositoryPath, commandFullPath));
+                        if (File.Exists(finalCommandPath))
+                        {
+                            commandFullPath = finalCommandPath;
+                        }
+                    }
+
+                    context.Logger.Log("Setting execute permissions for " + commandFullPath);
+                    PermissionHelper.Chmod("ugo+x", commandFullPath, Environment, DeploymentSettings, context.Logger);
```

Comment:
If it's a relative path and the file doesn't exist, won't you end up trying to `chmod` on that relative path here?

Reply:
Should we still do a file.Exists check before the chmod?

Reply:
Sure, this is all within the kudu container .. if the file doesnt exist, it will throw an error in the Kudu logs and VSCode displays that error which is good.

Reply:
Regardless of my comment above, I will make sure that the command is under the Repository path if the command is relative path.